File: Javascript/Algorithms/1502.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
let canMakeArithmeticProgression = arr => {
    arr.sort((a, b) => a-b);
    let step = arr[1] - arr[0],
        len = arr.length;
    for(let i = 2; i < len; i++){
        if (arr[i] != arr[i-1] + step)
            return false;
    }
    return true;  
};

File: Rust/Algorithms/1502.rs
impl Solution {
    pub fn can_make_arithmetic_progression(mut arr: Vec<i32>) -> bool {
        arr.sort_unstable();
        let step = arr[1] - arr[0];
        for i in 2..arr.len() {
            if arr[i] - arr[i - 1] != step {
                return false;
            }
        }
        true
    }
}
